src/v/datalake/translation/partition_translator.cc

```diff
@@ -284,13 +259,22 @@ ss::future<> partition_translator::run_one_translation_iteration(
         co_await std::move(translation_f).finally([&cancellation_timer] {
             cancellation_timer.cancel();
         });
-    } catch (...) {
-        ex = std::current_exception();
+        _inflight_translation_state->as.check();
+    } catch (const translator_out_of_memory_error&) {
         vlog(
           _logger.warn,
-          "Translation attempt failed: {}, discarding state to reset "
-          "translation",
-          ex);
+          "Translation attempt ran into OOM, result will be flushed "
+          "immediately");
+        force_flush_after_iteration = true;
+    } catch (const translator_time_quota_exceeded_error&) {
+        vlog(
+          _logger.debug,
+          "Translation attempt exceeded scheduler time limit quota");
```

Comment:
nit: maybe add an explicitly comment that we're not keeping the exceptions because the underlying state is still safe to use. Could be mistaken for an oversight

Reply:
done.